src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -37,11 +38,19 @@ public function process(ContainerBuilder $container)
             parent::process($container);
 
             foreach ($this->unusedBindings as list($key, $serviceId)) {
-                throw new InvalidArgumentException(sprintf('Unused binding "%s" in service "%s".', $key, $serviceId));
+                $message = sprintf('Unused binding "%s" in service "%s".', $key, $serviceId);
+                if ($this->errors) {
+                    $message .= sprintf("\nCould be related to%s:", 1 < \count($errors) ? ' one of' : '');
+                }
+                foreach ($this->errors as $serviceId => $mesage) {
+                    $message .= sprintf("\n - \"%s\": %s", $serviceId, $message);
```

Comment:
I don't think we need to put the service id here since it's already in the exception message thrown by `getConstructor`.